src/Mvc/Mvc.Razor/src/RazorPage.cs

```diff
@@ -18,12 +18,12 @@ public abstract class RazorPage : RazorPageBase
         private readonly HashSet<string> _renderedSections = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         private bool _renderedBody;
         private bool _ignoreBody;
-        private HashSet<string> _ignoredSections;
+        private HashSet<string>? _ignoredSections;
 
         /// <summary>
         /// An <see cref="HttpContext"/> representing the current request execution.
         /// </summary>
-        public HttpContext Context => ViewContext?.HttpContext;
+        public HttpContext Context => ViewContext?.HttpContext!;
```

Comment:
ViewContext can't be null

Reply:
In practice no. But in theory, if you're unit testing, you could have chosen to not initialize it. We want to avoid producing a null ref from the framework code in that case. 

```C#
var page = new RazorPage();
page.HttpContext // fine if it's null here, but we want to avoid a NullReferenceException
```